com.unity.render-pipelines.core/Editor/ContextualMenuDispatcher.cs

```diff
@@ -54,99 +54,91 @@ static void RemoveCameraComponent(MenuCommand command)
             RemoveComponent<Camera>(command);
         }
 
-        [InitializeOnLoadMethod]
-        static void RegisterAdditionalDataMenus()
+        /// <summary>
+        /// Removes a <see cref="IAdditionalData"/> and it's components defined by <see cref="RequireComponent"/>
+        /// </summary>
+        /// <typeparam name="T">A <see cref="MonoBehaviour"/> that is an <see cref="IAdditionalData"/></typeparam>
+        /// <exception cref="Exception">If the given <see cref="MonoBehaviour"/> is not an <see cref="IAdditionalData"/></exception>
+        public static void RemoveAdditionalData<T>(MenuCommand command)
+            where T : Component, IAdditionalData
         {
-            foreach (var additionalData in TypeCache.GetTypesDerivedFrom<IAdditionalData>())
+            if (typeof(T).GetCustomAttributes(typeof(RequireComponent), true).FirstOrDefault() is RequireComponent rc)
```

Comment:
`RequireComponent` attribute can be applied multiple times to a type, maybe you can use a foreach to execute the code over all atribute instances?

Reply:
Applied on `TryGetComponentsToRemove`